talk/owt/sdk/p2p/p2pclient.cc

```diff
@@ -251,79 +256,88 @@ void P2PClient::OnSignalingMessage(const std::string& message,
           << "Chat cannot be setup since the remote user is not allowed.";
       return;
     }
+    Json::Reader reader;
+    Json::Value json_message;
+    if (!reader.parse(message, json_message)) {
+      RTC_LOG(LS_WARNING) << "Cannot parse incoming message.";
+      return;
+    }
+    std::string message_type;
+    rtc::GetStringFromJsonObject(json_message, kMessageTypeKey, &message_type);
     if (!that->IsPeerConnectionChannelCreated(remote_id)) {
-      if (message.find("\"type\":\"chat-closed\"") != std::string::npos) {
+      if (message_type == "chat-closed") {
```

Comment:
please move all the protocol related literals, like "chat-closed", "offer" to the const string listing part of this file.

Reply:
Done.